### ISSUE TITLE
Contributing to Apache Arrow

### ISSUE THREAD

**Issue by wesm** [jihoonson/arrow-rs#1]
hi @jihoonson -- what would you think about contributing your Rust bindings to the Arrow project itself? We would love to have you involved.

**Comment by jihoonson** [comment]
@wesm, thanks for your suggestion. It would be great. 
I found some issues while building against the recent build of Arrow. I'll fix it before contributing.

Would you let me know what should I do? Is it ok to just make a PR?

**Comment by wesm** [comment]
Yes, that would be great. It may be interesting to start a native in-memory Arrow implementation for Rust -- it may be simpler to use the C++ library for messaging and IO

**Comment by jihoonson** [comment]
Yes, it is desired. I tested the current implementation before and it was not good because some frequently called functions are not inlined due to ffi.

**Comment by sadikovi** [comment]
@jihoonson Is the project part of the Apache Arrow repository now? Do you accept contributions to this project? If so, could you share the roadmap - I would like to contribute. Thanks!

**Comment by wesm** [comment]
It seems that a native Rust generator for Flatbuffers isn't yet available: https://github.com/google/flatbuffers/pull/3894. One could interact with the Arrow metadata via C, though. It would probably be better to make a native Arrow implementation in Rust rather than wrapping the C++, but let me know what others think. Happy to have you involved in the community

**Comment by sadikovi** [comment]
@wesm Thank you for the clarification! Meanwhile, I will try building the project and see if there is anything missing that I can add. @jihoonson - I would appreciate any direction!

**Comment by jihoonson** [comment]
Hi @sadikovi, this project is not a part of Apache Arrow yet. Actually, I'm working on a native Rust implementation of Arrow (https://github.com/jihoonson/iron-arrow), but it's still very beginning and not mature enough. I'm thinking to donate it to Apache Arrow once its basic structure (like Array or RecordBatch) is implemented. Welcome any contributions!

**Comment by sadikovi** [comment]
@jihoonson thanks!

**Comment by wesm** [comment]
I'm going to try to learn some Rust and help with this once I know enough, it would be great if we can bootstrap an implementation in Apache Arrow to simplify governance and IP lineage

**Comment by sadikovi** [comment]
@wesm that is great! I opened pull request to resolve conflicts on Rust PR in flatbuffers, to push that work forward somehow. I do not quite understand where to start:) - would appreciate any guidance on what to work on.

**Comment by jihoonson** [comment]
@wesm good to hear that. I restarted working on the native implementation recently, and try to do my best to finish some basic implementation as soon as possible.

**Comment by sunchao** [comment]
Would love to contribute to native Rust impl for Arrow too!
